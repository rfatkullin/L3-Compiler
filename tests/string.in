func AtoB(str -> [char]; len -> int) -> [char]
	i -> int;

	for i = 0 to len - 1 do
		if (str[i] == 'A') then
			str[i] = 'B';
		elseif (str[i] == 'a') then
			str[i] = 'b';
		endif;
	endfor;

	AtoB = str;
endfunc

func main (args->[[char]]) -> int
	str -> [char];

	str = "Azaza";

	print(AtoB(str, 5));

	print(" ");

	str1 = "Hello$@$Hello" -> [char];
	str2 = "Hello$<$Hello" -> [char];
	str3 = "Hello$>$Hello" -> [char];
	str4 = "Hello$^$Hello" -> [char];
	str5 = "Hello$<<$Hello" -> [char];
	str6 = "Hello$$Hello" -> [char];
	str7 = "Hello$"$Hello" -> [char];
	str8 = "Hello$'$Hello" -> [char];

	str9 = "Hello$7$Hello" -> [char];
	str10 = "Hello$8$Hello" -> [char];
	str11 = "Hello$9$Hello" -> [char];
	str12 = "Hello$A$Hello" -> [char];
	str13 = "Hello$D$Hello" -> [char];
	str14 = "Hello$24$Hello" -> [char];
	str15 = "Hello$22$Hello" -> [char];
	str16 = "Hello$27$Hello" -> [char];

	print(str1);
	print(str2);
	print(str3);
	print(str4);
	print(str5);
	print(str6);
	print(str7);
	print(str8);

	print(" ");

	print(str9);
	print(str10);
	print(str11);
	print(str12);
	print(str13);
	print(str14);
	print(str15);
	print(str16);

	main = 0;
endfunc
