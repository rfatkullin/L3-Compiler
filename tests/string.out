Bzbzb
 
HelloHello
HelloHello
Hello	Hello
Hello
Hello
HelloHello
HelloHello
Hello"Hello
Hello'Hello
 
HelloHello
HelloHello
Hello	Hello
Hello
Hello
HelloHello
HelloHello
Hello"Hello
Hello'Hello
